1.1.0 (#471)
odoo/odoo-ls: changelog.md

# Changelog

## [1.1.0] - 2025/11/06 - Workspace Symbols / WSL support

> This Beta update improve the QoL on various IDE and bring some new features:

Reviewer: ```suggestion
This Beta update improves the QoL on various IDEs and brings some new features:
```

- Workspace Symbol Lookup allows you to search for any class/function/model/xml_id in the whole project (ctrl-t on vscode). All xml_id are prefixed
by `xmlid.` and model names are quoted.

![workspace symbol lookup image](https://github.com/odoo/odoo-ls/blob/images-changelogs/images/workspace_symbol_lookup.png?raw=true)

- Import statements now have autocompletion, hover and gotodefinition features.
- We now have a better support for WSL paths!

## VsCode

- Better display about the server status.

## PyCharm

- Improve the lifecycle of OdooLS. Server will be always running but idle, and starts only if a configuration is detected. It implies
that we removed the 'start server' button as it is not useful anymore. It should end up in a more clear and usable interface.

> - Fix the "Disabled" profil behavior that was preventing any further profil change.

Reviewer: ```suggestion
- Fix the "Disabled" profile behavior that was preventing any further profile change.
```

> - Starting from 2025.3, PyCharm will be able to display the loading status of the server

Reviewer: ```suggestion
- Starting from 2025.3, PyCharm will be able to display the loading status of the server.
```

- Attach additional stubs (with lxml) to the build
- Add configuration wiki link on the settings page.
- Update deprecated API calls to isAarch64 methods.

## Server

- Support for workspace Symbol requests.

> - Core structure to support `$/Cancel` notification and use it for Workspace Symbol Request. These notifications indicates that a job can be cancelled because no more useful.

Reviewer: ```suggestion
- Core structure to support `$/cancelRequest` notification and use it for Workspace Symbol Request. These notifications indicates that a job can be cancelled because no more useful.
```

- Import statements now have autocompletion, hover feature and gotodefinition.
- Server now supports WSL paths, including `file:////` or `file://wsl.localhost`
- Server now use workDoneProgress to report loading status to client. Client that supports this feature will now display the loading progression at startup.
- Server will now send more information about its status: it can indicates if it is waiting for a git lock to be freed.
- Autocompletion and validation for inverse_name keyword argument.

> - Crash reports will now include lasts LSP messages to help the debugging and give us a better overview of what happen before the crash

Reviewer: ```suggestion
- Crash reports will now include the latest LSP messages to help the debugging and give us a better overview of what happened before the crash.
```

- It is now possible to autocomplete slices ( `self.env["`) even without closing the brackets

> - Diagnostic filters in configuration files can now accept variables like `${userHome}` or `${workspaceFolder}`

Reviewer: ```suggestion
- It is now possible to autocomplete slices ( `self.env["`) even without closing the brackets.
- Diagnostic filters in configuration files can now accept variables like `${userHome}` or `${workspaceFolder}`.
```

> - You can now hover and gotodefinition for module name in `__manifest__.py` files.

Reviewer: ```suggestion
- You can now hover and gotodefinition for module names in `__manifest__.py` files, and in hover you could also see full list of the module's dependencies.
```

> - `filtered` and `filtered_domain` now has a proper return value.

Reviewer: ```suggestion
- `filtered` and `filtered_domain` now have a proper return value.
```

- Remove diagnostics of ImportError in the `except` block of a `try..except ImportError` statement.

> - Doing a gotodefinition on a `display_name` will now redirect you to the compute method

Reviewer: ```suggestion
- Doing a gotodefinition on a `display_name` will now redirect you to the compute method(s).
```

- Update Ruff dependencies to 0.14.3

### Fixes

- Doing a GoToDefinition on a string will not lead you to the 'string' definition anymore. 'String' definition is not the definition of the current string, but it's type.

> - Doing a cyclic dependency between 'modules depends' will not crash anymore but generate the diagnostic OLS04012

Reviewer: ```suggestion
- Doing a cyclic dependency between 'modules depends' will not crash anymore but generate the diagnostic OLS04012.
```

- Fix the path to additional stubs and so fix the usage of the lxml stubs.
- If odoo_path is ending with `.something` but is pointing to a valid directory, the server should not consider `something` as a file extension, but as a part of the folder name
- Fix some internal hooks to work with user defined Fields, instead of only the Odoo ones.
- Configuration option "diag_missing_imports" is now really taken into account when generating diagnostics about imports.

> - Fix dependencies on comodel and relation fields diagnostics

Reviewer: ```suggestion
- Fix dependencies on comodel and relation fields diagnostics.
```

- Fix missing ImportError diagnostic on import statement without a 'from' or 'as' part.

> - Various small fixes and typo

Reviewer: ```suggestion
- Various small fixes and typos.
```
